Minor improvement to `qtbot` testing section (#481)
napari/docs: docs/developers/contributing/testing.md

recommend that you use the following fixtures when needing a widget or
{class}`~napari.Viewer` in a test.

#### qapp and qtbot

> If you need a `QApplication` to be running for your test you can use the

Reviewer: ```suggestion
If you need to use any Qt related code in your test you need to ensure that `QApplication` is created. To do this we suggest use  
```

[`qapp`](https://pytest-qt.readthedocs.io/en/latest/reference.html#module-pytestqt.plugin)
fixture from [`pytest-qt`](https://pytest-qt.readthedocs.io/en/latest/index.html),
a napari testing dependency.

If you need to instantiate a Qt object (e.g., a widget) for your test, we recommend

> that you use the

Reviewer: ```suggestion
If you need to instantiate a Qt GUI object (e.g., a widget) for your test, we recommend
that you use the
```

[`qtbot`](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot)
fixture. `qtbot`, which itself depends on `qapp` , allows you to test user input
(e.g., mouse clicks) by sending events to Qt objects.

````{note}
Fixtures in pytest can be a little mysterious, since it's not always
clear where they are coming from.  The `pytest-qt` `qapp` and `qtbot` fixtures
can be used in two ways; by adding them to the list of arguments of your test function:

```python
def test_something(qtbot):
    ...
```
or by using pytest's `usefixtures`, which avoids adding an unused argument to your
test function:

```python
@pytest.mark.usefixtures('qtbot')
def test_something():
    ...
```
<br/>
````

`qtbot` also provides a convenient

> [`addWidget`](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot.addWidget)

Reviewer: ```suggestion
[`add_widget`/`addWidget`](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot.addWidget)
```

Author: Do you want both or is there one that you prefer? I guess `add_widget` would be more 'python-ic'?

Author: Looks okay as both so will leave

Reviewer: In our test we are using both of them. So I would like to signal that both are valid.

> method that will ensure that the widget gets closed at the end of the test.

Reviewer: ```suggestion
method that will ensure that the widget gets closed and properly cleaned at the end of the test. This prevents fropm random segfaults in tests run. 
```

This prevents leakage between tests.
The

> [`waitUntil`](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot.waitUntil) method is also useful to wait for a desired condition. The example below

Reviewer: ```suggestion
[`wait_until`/`waitUntil`](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot.waitUntil) method is also useful to wait for a desired condition. The example below
```
